### PR TITLE
Rvinnakota/us111803 annotations can be added to an assignment

### PR DESCRIPTION
**Pull request by rohitvinnakota**
LMS PR here:
https://git.dev.d2l/projects/CORE/repos/lms/pull-requests/5925/overview

Rally story here:
https://rally1.rallydev.com/#/110294864140d/detail/userstory/349211942424

PR in AssignmentEditorDetail in BrightspaceHypermediaComponents/activities to follow

### CODE REVIEW

File: src/activities/assignments/AssignmentEntity.js [hunk 1]
```diff
@@ -258,4 +258,28 @@ export class AssignmentEntity extends Entity {
 		];
 		await performSirenAction(this._token, action, fields);
 	}
+
+	annotationToolsAvailable() {
```
Comment (ryantmer) [inline src/activities/assignments/AssignmentEntity.js]: I'd recommend `getAvailableAnnotationTools` here - just to keep consistent the `get` nomenclature elsewhere

File: src/activities/assignments/AssignmentEntity.js [hunk 2]
```diff
@@ -258,4 +258,28 @@ export class AssignmentEntity extends Entity {
...
+		return this._entity.getSubEntityByClasses(['annotations', 'enabled']);
```
Comment (ryantmer) [inline src/activities/assignments/AssignmentEntity.js]: I'm not 100% sure, but I _suspect_ you actually want `getSubEntitiesByClasses` (plural) here? `getSubEntityByClasses` (singular) will just return the first entity that matches the requirements, whereas `getSubEntitiesByClasses` (plural) will return an array of all entities that match the requirements.

File: src/activities/assignments/AssignmentEntity.js [hunk 3]
```diff
@@ -258,4 +258,28 @@ export class AssignmentEntity extends Entity {
...
+	}
+
+	canSeeAnnotations() {
```
Comment (ryantmer) [inline src/activities/assignments/AssignmentEntity.js]: Adding a quick JSDoc to each of these would be appreciated! We're not perfect on it, but all new things should have them.

File: src/activities/assignments/AssignmentEntity.js (hunk 2 above)
Comment (ryantmer) [inline src/activities/assignments/AssignmentEntity.js]: Applies elsewhere too, but there is a `Classes` in `hypermedia-constants` that these could be added to.

Comment (rohitvinnakota) [reply src/activities/assignments/AssignmentEntity.js]: I just want the single `annotations` sub-entity that contains the action and the current state of annotations for the assignment. There should be only one sub-entity with an `annotations` class per assignment.

File: src/activities/assignments/AssignmentEntity.js (hunk 1 above)
Comment (rohitvinnakota) [reply src/activities/assignments/AssignmentEntity.js]: Is `_getAvailableAnnotationTools` preferred over `getAvailableAnnotationTools`?

Comment (ryantmer) [reply src/activities/assignments/AssignmentEntity.js]: Depends - is this being used by your consumer as well? (The underscore is just being used to denote private methods by convention, really.)

Comment (rohitvinnakota) [reply src/activities/assignments/AssignmentEntity.js]: Yes it is to set the checkbox state, so `getAvailableAnnotationTools`?

Comment (ryantmer) [reply src/activities/assignments/AssignmentEntity.js]: Sounds right to me!

File: src/activities/assignments/AssignmentEntity.js (hunk 2 above)
Comment (ppaskaris-d2l) [reply src/activities/assignments/AssignmentEntity.js]: Shouldn't we ideally look for things by relationship and not their class?

Comment (ryantmer) [reply src/activities/assignments/AssignmentEntity.js]: Actually, that's a better point. `getSubEntityByClass` can generally be replaced with a call to `getSubEntityByRel`.

**Pull request merged**